crates/payload/builder/src/payload.rs

```diff
@@ -245,19 +245,39 @@ impl PayloadBuilderAttributes for EthPayloadBuilderAttributes {
             })
             .map(BlobExcessGasAndPrice::new);
 
+        let basefee = parent
+            .next_block_base_fee(chain_spec.base_fee_params(self.timestamp()))
+            .or_else(|| {
+                if chain_spec.fork(Hardfork::London).transitions_at_block(parent.number + 1) {
+                    // default base fee is zero
+                    Some(EIP1559_INITIAL_BASE_FEE)
+                } else {
+                    None
+                }
+            })
+            .map(U256::from);
+
+        // If we are on the London fork boundary, we need to multiply the parent's gas limit by the
+        // elasticity multiplier to get the new gas limit.
+        let gas_limit = if chain_spec.fork(Hardfork::London).transitions_at_block(parent.number + 1)
```

Comment:
I guess we kinda have to check this always,
but we could combine this with the basefee call above, though it may not worth it

Reply:
decided to combine